File: src/main/java/main/utils/apis/robertify/imagebuilders/ImageType.java
package main.utils.apis.robertify.imagebuilders;

import java.util.Arrays;
import java.util.List;

public enum ImageType {
    NOW_PLAYING("music", "nowplaying"),
    QUEUE("music", "queue");

    private final String[] segments;

    ImageType(String... segments) {
        this.segments = segments;
    }

    public List<String> getSegments() {
        return Arrays.stream(segments).toList();
    }
}

File: src/main/kotlin/main/utils/api/robertify/imagebuilders/ImageType.kt
package main.utils.api.robertify.imagebuilders

enum class ImageType(private vararg val segments: String) {
    NOW_PLAYING("music", "nowplaying"),
    QUEUE("music", "queue"),
    PLAYLISTS_LIST("playlists", "list"),
    PLAYLISTS_CONTENT("playlists", "content");

    fun getSegments(): List<String> = segments.toList()

}